packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -140,7 +140,19 @@ interface RootedProxyPaths {
 export function prepareContentForInsert(
 	content: InsertableContent,
 	forest: IForestSubscription,
-): FactoryContent {
+): FactoryContent;
+export function prepareContentForInsert(
+	content: InsertableContent | undefined,
+	forest: IForestSubscription,
+): FactoryContent | undefined;
+export function prepareContentForInsert(
+	content: InsertableContent | undefined,
```

Comment:
It's only objects that allow undefined. ArrayNodes and MapNodes do not - you can't have undefined as an inserted array element, and you can't set a map value to undefined. Which is why there are some usages of these functions that never take undefined and never return undefined, and some that do both - hence the overloads.

Reply:
Right, makes sense. Should we consider adding an `ObjectInsertableContent` type or something? No strong feelings here, just curious if you think that might be valuable or help clarify things for users of the APIs.

Reply:
It's a good question. When it comes to undefined, I generally like to see `X | undefined` rather than `XOrUndefined` because `undefined` is such a special value and type. Though, I could see an argument for doing `XOrUndefined` in this particular case because it's a little unusual - we _are_ using `undefined` as an actual value, that we examine and then which causes a mutation to "nothing". But, from the user's perspective, it's the same as just JS `= undefined`. I don't feel strongly either way either.

Reply:
I don't think it's `XOrUndefined`, it's an item with its own semantics, specifically the kinds of data insertable into an object field. As an implementation detail, that happens to be "more generally insertable data" | undefined, but I could see that potentially changing in the future.

But I do agree that aliasing away the `| undefined` is generally more disruptive than helpful.